static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx

```diff
@@ -24,215 +24,209 @@ import type {
-  codeMappings: RepositoryProjectPathConfig[];
-  codeownersFile: CodeownersFile | null;
-  error: boolean;
-  errorJSON: {raw?: string} | null;
-  integrations: Integration[];
-  isLoading: boolean;
-} & DeprecatedAsyncComponent['state'];
+type TCodeownersPayload = {codeMappingId: string | null; raw: string};
+type TCodeownersData = CodeOwner;
+type TCodeownersError = RequestError;
+type TCodeownersVariables = [TCodeownersPayload];
+type TCodeownersContext = unknown;
 
-export default class AddCodeOwnerModal extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      codeownersFile: null,
-      codeMappingId: null,
-      isLoading: false,
-      error: false,
-      errorJSON: null,
-    };
-  }
+export default function AddCodeOwnerModal({
+  organization,
+  Header,
+  Body,
+  Footer,
+  project,
+  onSave,
+  closeModal,
+}: Props) {
+  const {
+    data: codeMappings,
+    isPending: isCodeMappingsPending,
+    isError: isCodeMappingsError,
+  } = useApiQuery<RepositoryProjectPathConfig[]>(
+    [
+      `/organizations/${organization.slug}/code-mappings/`,
+      {query: {project: project.id}},
+    ],
+    {staleTime: Infinity}
+  );
 
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization, project} = this.props;
-    const endpoints: ReturnType<DeprecatedAsyncComponent['getEndpoints']> = [
-      [
-        'codeMappings',
-        `/organizations/${organization.slug}/code-mappings/`,
-        {query: {project: project.id}},
-      ],
-      [
-        'integrations',
-        `/organizations/${organization.slug}/integrations/`,
-        {query: {features: ['codeowners']}},
-      ],
-    ];
-    return endpoints;
-  }
+  const {
+    data: integrations,
+    isPending: isIntegrationsPending,
+    isError: isIntegrationsError,
+  } = useApiQuery<Integration[]>(
+    [
+      `/organizations/${organization.slug}/integrations/`,
+      {query: {features: ['codeowners']}},
+    ],
+    {staleTime: Infinity}
+  );
 
-  fetchFile = async (codeMappingId: string) => {
-    const {organization} = this.props;
-    this.setState({
-      codeMappingId,
-      codeownersFile: null,
-      error: false,
-      errorJSON: null,
-      isLoading: true,
-    });
-    try {
-      const data: CodeownersFile = await this.api.requestPromise(
-        `/organizations/${organization.slug}/code-mappings/${codeMappingId}/codeowners/`,
-        {
-          method: 'GET',
-        }
-      );
-      this.setState({codeownersFile: data, isLoading: false});
-    } catch (_err) {
-      this.setState({isLoading: false});
-    }
-  };
+  const [codeMappingId, setCodeMappingId] = useState<string | null>(null);
 
-  addFile = async () => {
-    const {organization, project, onSave, closeModal} = this.props;
-    const {codeownersFile, codeMappingId, codeMappings} = this.state;
+  const {
+    data: codeownersFile,
+    isPending: isCodeownersFilePending,
```

Comment:
Note that this will be `true` whenever this query is disabled. If that's not what you want, you should use `isLoading` instead